mav_nonlinear_mpc/src/nonlinear_mpc.cc

```diff
@@ -522,6 +525,19 @@ bool NonlinearModelPredictiveControl::getPredictedState(
   for (size_t i = 0; i < ACADO_N + 1; i++) {
     mav_msgs::EigenTrajectoryPoint pnt;
     pnt.position_W = state_.block(i, 6, 1, 3).transpose();
+    pnt.velocity_W = state_.block(i, 0, 1, 3).transpose();
+
+    tf::Quaternion tf_orientation;
+    tf_orientation.setRPY(state_(i,3), state_(i,4), state_(i,5));
+    pnt.orientation_W_B.x() = tf_orientation.x();
+    pnt.orientation_W_B.y() = tf_orientation.y();
+    pnt.orientation_W_B.z() = tf_orientation.z();
+    pnt.orientation_W_B.w() = tf_orientation.w();
+                
+    pnt.time_from_start_ns = static_cast<int64_t>(i+1) *
```

Comment:
why `i+1` ? the first state x_0 is the current state not one time step ahead.

Reply:
It should be just i then, I didn't realize the current state was an output of the solver